pkg/compiler/analysis.go

```diff
@@ -40,6 +40,22 @@ func (c *codegen) getIdentName(pkg string, name string) string {
 // and returns number of variables initialized and
 // true if any init functions were encountered.
 func (c *codegen) traverseGlobals() (int, bool) {
+	var hasDefer bool
+	c.ForEachFile(func(f *ast.File, _ *types.Package) {
+		ast.Inspect(f, func(node ast.Node) bool {
+			if hasDefer {
+				return false
+			} else if _, ok := node.(*ast.DeferStmt); ok {
+				hasDefer = true
+				return false
+			}
+			return true
+		})
+	})
+	if hasDefer {
+		c.globals["<exception>"] = exceptionIndex
+		c.globals["<finally_state>"] = finallyProcessedIndex
```

Comment:
Why can't we use local variables here? Both catch and finally are to be executed with the same local scope (function's scope).

Reply:
Because panic can occur in a called function and handled somewhere outside.

Reply:
But this "outside" will get en exception (panic string in our case) via parameter and it will have its function local context available.

Reply:
Oh, this was only about the `finallyProcessedIndex`. Fixed.